api/src/components/reference/service.ts

```diff
@@ -17,18 +18,43 @@ export const getAllByPublication = async (publicationId: string) => {
         }
     });
 
-    return references;
+    // Return with publicationId.
+    return references.map((reference) => {
+        const { publicationVersionId, ...referenceRest } = reference;
+
+        return { publicationId, ...referenceRest };
+    });
 };
 
-export const updateAll = async (publicationVersionId, data) => {
+export const updateAll = async (publicationId, data: I.UpdateReferencesBody) => {
+    const latestVersion = await client.prisma.publicationVersion.findFirst({
+        where: {
+            versionOf: publicationId,
+            isCurrent: true
+        },
+        select: {
+            id: true
+        }
+    });
+
+    if (!latestVersion) {
+        throw Error('Publication does not have a latest version');
+    }
+
     await client.prisma.references.deleteMany({
         where: {
-            publicationVersionId
+            publicationVersionId: latestVersion?.id
         }
     });
 
+    // Event body contains a publication ID. Replace this with the latest version's ID.
+    const createManyInput = data.map((reference) => {
```

Comment:
Instead of doing all these spreads while looping, just to keep the UI happy, i think we should simply change the "publicationId" field with "publicationVersionId" on the UI side. This will also reflect what we have into DB